tsweb/tsweb.go

```diff
@@ -502,23 +518,29 @@ func (h logHandler) logRequest(r *http.Request, lw *loggingResponseWriter, msg A
 	msg.Bytes = lw.bytes
 	msg.Seconds = h.opts.Now().Sub(msg.Time).Seconds()
 	switch {
+	case msg.Code != 0:
+		// Keep explicit codes from a few particular errors.
 	case lw.hijacked:
 		// Connection no longer belongs to us, just log that we
 		// switched protocols away from HTTP.
 		msg.Code = http.StatusSwitchingProtocols
 	case lw.code == 0:
-		if r.Context().Err() != nil {
-			// We didn't write a response before the client disconnected.
-			msg.Code = 499
-		} else {
-			// If the handler didn't write and didn't send a header, that still means 200.
-			// (See https://play.golang.org/p/4P7nx_Tap7p)
-			msg.Code = 200
-		}
+		// If the handler didn't write and didn't send a header, that still means 200.
+		// (See https://play.golang.org/p/4P7nx_Tap7p)
+		msg.Code = 200
 	default:
 		msg.Code = lw.code
 	}
 
+	// Keep track of the original response code when we've overridden it.
+	if lw.code != 0 && msg.Code != lw.code {
+		if msg.Err == "" {
+			msg.Err = fmt.Sprintf("(returned code %d)", lw.code)
+		} else {
+			msg.Err = fmt.Sprintf("%s (returned code %d)", msg.Err, lw.code)
```

Comment:
Not sure if "returned code" makes it clear that this is the response code we intended to return (and maybe even wrote to the client?) but the connection got closed before we could finish.

I don't have a much better name, but some ideas are: "response code", "original code" or maybe "intended status code"? 🤷

Reply:
I've gone with "original code"